js-envs/src/main/scala/org/scalajs/jsenv/ExternalJSEnv.scala

```diff
@@ -25,9 +25,22 @@ abstract class ExternalJSEnv(
   /** Custom initialization scripts. */
   protected def customInitFiles(): Seq[VirtualJSFile] = Nil
 
-  protected class AbstractExtRunner(protected val classpath: CompleteClasspath,
-      protected val code: VirtualJSFile, protected val logger: Logger,
-      protected val console: JSConsole) {
+  protected class AbstractExtRunner(
+      protected val libs: Seq[ResolvedJSDependency],
+      protected val sjsCode: VirtualJSFile,
+      protected val code: VirtualJSFile) {
```

Comment:
The `NodeJSEnv` for example makes a difference between `sjsCode` and `code` (how it loads it). Conceptually, `sjsCode` is code that may not "do" anything. I.e. it may only define stuff. Further, there is a symmetry with the `LinkingUnitJSEnv` if we have two params.